### PR TITLE
Pulsar fix performance issue

### PR DESCRIPTION
**Pull request by rshermanTHG**
When reviewing the code again I noticed that the PulsarProducer class was creating a new connection to the Pulsar Broker for each message sent. This PR is mainly to fix that issue. It additionally fixes an issue where the producer name was hardcoded and ignored any configured name.

I will be creating another PR against camel-2.x later today.

### CODE REVIEW

File: components/camel-pulsar/src/main/java/org/apache/camel/component/pulsar/PulsarProducer.java [hunk 1]
```diff
@@ -16,57 +16,72 @@
...
     private final PulsarEndpoint pulsarEndpoint;
+    private Producer<byte[]> producer;
 
     public PulsarProducer(PulsarEndpoint pulsarEndpoint) {
         super(pulsarEndpoint);
-
         this.pulsarEndpoint = pulsarEndpoint;
     }
 
+    public static PulsarProducer create(final PulsarEndpoint pulsarEndpoint) {
```
Comment (oscerd) [inline components/camel-pulsar/src/main/java/org/apache/camel/component/pulsar/PulsarProducer.java]: This is never used as far as I see.

Also the producer is already created in the PulsarEndpoint.

https://github.com/apache/camel/blob/master/components/camel-pulsar/src/main/java/org/apache/camel/component/pulsar/PulsarEndpoint.java#L60

Comment (rshermanTHG) [reply components/camel-pulsar/src/main/java/org/apache/camel/component/pulsar/PulsarProducer.java]: Sorry an issue with cherry-picking a commit has brought in some deleted code will re-remove

File: components/camel-pulsar/src/main/java/org/apache/camel/component/pulsar/PulsarProducer.java [hunk 2]
```diff
@@ -16,57 +16,72 @@
...
-import java.util.Map;
-
 import org.apache.camel.Exchange;
 import org.apache.camel.Message;
 import org.apache.camel.NoTypeConversionAvailableException;
 import org.apache.camel.TypeConversionException;
-import org.apache.camel.component.pulsar.utils.message.PulsarMessageHeaders;
 import org.apache.camel.component.pulsar.utils.message.PulsarMessageUtils;
-import org.apache.camel.support.DefaultProducer;
+import org.apache.camel.impl.DefaultProducer;
```
Comment (rshermanTHG) [inline components/camel-pulsar/src/main/java/org/apache/camel/component/pulsar/PulsarProducer.java]: Will revert this change
